src/test/java/de/adorsys/keycloak/config/service/SyncUserFederationIT.java

```diff
@@ -0,0 +1,117 @@
+/*-
+ * ---license-start
+ * keycloak-config-cli
+ * ---
+ * Copyright (C) 2017 - 2021 adorsys GmbH & Co. KG @ https://adorsys.com
+ * ---
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ---license-end
+ */
+
+package de.adorsys.keycloak.config.service;
+
+import de.adorsys.keycloak.config.AbstractImportTest;
+import org.hamcrest.core.IsNot;
+import org.hamcrest.core.IsNull;
+import org.junit.internal.matchers.ThrowableMessageMatcher;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.keycloak.representations.idm.RealmRepresentation;
+import org.springframework.test.context.TestPropertySource;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+
+@TestPropertySource(properties = {
+        "import.sync-user-federation=true"
+})
+public class SyncUserFederationIT extends AbstractImportTest {
+
+    private static final String REALM_NAME = "realmWithLdap";
+    private static final String REALM_NAME_WITHOUT_FEDERATION = "realmWithoutLdap";
+    private static final String FUNCTION_WHERE_THE_ERROR_MUST_BE_GENERATED = "syncUserFederationIfNecessary";
+
+    public SyncUserFederationIT() {
+        this.resourcePath = "import-files/user-federation";
+    }
+
+    @Test
+    @Order(0)
+    void shouldCreateRealmWithUser() {
+        Throwable throwable = null;
+        try {
+            doImport("00_create_realm_with_federation.json");
+        } catch (Throwable t) {
+            throwable = t;
+        }
+
+        assertThat(throwable, new IsNot<>(new IsNull<>()));
+        // This matching use the name of the function where the error occurs in order to guarantee the source.
+        boolean throwableCameFromUserFederationSync = Arrays.stream(throwable.getStackTrace())
+                .anyMatch(stackTraceElement -> stackTraceElement.getMethodName().contains(FUNCTION_WHERE_THE_ERROR_MUST_BE_GENERATED));
+        assertThat(throwable, new ThrowableMessageMatcher<>(is("HTTP 500 Internal Server Error")));
+        assertThat(throwableCameFromUserFederationSync, is(true));
```

Comment:
One question here: 

Why assertThrows wont fit here? Like https://github.com/adorsys/keycloak-config-cli/blob/e9cefdcfba9a38aaf4e38e1ef15f399b7bca04d9/src/test/java/de/adorsys/keycloak/config/service/ImportSimpleRealmIT.java#L196-L202

Reply:
It will fit perfectly. I will make the modification today.